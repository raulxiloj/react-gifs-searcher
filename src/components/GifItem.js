import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {

    return (
        <div className="col-md-3 mb-4 align-items-stretch animate__animated animate__fadeIn">
            <div className="card" >

                <img className="card-img-top" src={url} alt={title} />

                <div className="card-body">
                    <p className="card-text">{title}</p>
                </div>

            </div>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}