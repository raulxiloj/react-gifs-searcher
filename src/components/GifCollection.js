import React from 'react'
import PropTypes from 'prop-types';
import { GifItem } from './GifItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifCollection = ({ category }) => {//Gif grid

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    images.map(img => <GifItem key={img.id} {...img} />)
                }
            </div>
        </>
    )
}

GifCollection.propTypes = {
    category: PropTypes.string.isRequired
}