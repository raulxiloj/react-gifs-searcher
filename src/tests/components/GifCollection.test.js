import React from 'react';
import { shallow } from "enzyme"
import { GifCollection } from "../../components/GifCollection"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Tests for GifCollection component ', () => {

    const category = 'Hello'

    test('It should render the component correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifCollection category={category} />)
        expect(wrapper).toMatchSnapshot();
    })

    test('It should show the items when the imgs are loaded useFetchGifs', () => {

        const gifs = [{
            id: '123',
            title: 'Anything gg',
            url: 'https://localhost/anything',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifCollection category={category} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifItem').length).toBe(gifs.length);
    })


})
