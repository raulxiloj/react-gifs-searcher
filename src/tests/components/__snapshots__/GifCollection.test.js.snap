// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifCollection component  It should render the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hello
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Tests for GifCollection component  It should show the items when the imgs are loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hello
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifItem
      id="123"
      key="123"
      title="Anything gg"
      url="https://localhost/anything"
    />
  </div>
</Fragment>
`;
